File: app/src/main/java/de/gaffga/android/zazentimer/service/SectionEndReceiver.java
package de.gaffga.android.zazentimer.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

public class SectionEndReceiver extends BroadcastReceiver {
    private static final String TAG = "ZMT_SectionEndReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive: action=" + intent.getAction());
        Intent serviceIntent = new Intent(context, MeditationService.class);
        serviceIntent.setAction(MeditationService.ACTION_SECTION_ENDED);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(serviceIntent);
        } else {
            context.startService(serviceIntent);
        }
    }
}

File: app/src/main/kotlin/at/priv/graf/zazentimer/service/SectionEndReceiver.kt
package at.priv.graf.zazentimer.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class SectionEndReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive: action=${intent.action}")
        val serviceIntent = Intent(context, MeditationService::class.java)
        serviceIntent.action = MeditationService.ACTION_SECTION_ENDED
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(serviceIntent)
        } else {
            context.startService(serviceIntent)
        }
    }

    companion object {
        private const val TAG = "ZMT_SectionEndReceiver"
    }
}
